### PR TITLE
test: enable code coverage

### PR DESCRIPTION
**Pull request by aahna-ashina**
https://app.dework.xyz/nation3/develop-1?taskId=f67456db-3e0b-4d07-a699-e971294c0ecb

```
forge coverage
[⠒] Compiling...
[⠆] Compiling 21 files with 0.8.13
[⠒] Solc 0.8.13 finished in 7.30s
Compiler run successful
Analysing contracts...
Running tests...
+---------------------------------------------------+-----------------+-----------------+----------------+-----------------+
| File                                              | % Lines         | % Statements    | % Branches     | % Funcs         |
+==========================================================================================================================+
| script/CollateralAgreementDeployer.s.sol          | 0.00% (0/5)     | 0.00% (0/7)     | 100.00% (0/0)  | 0.00% (0/1)     |
|---------------------------------------------------+-----------------+-----------------+----------------+-----------------|
| src/agreements/CollateralAgreement.sol            | 100.00% (66/66) | 100.00% (86/86) | 50.00% (19/38) | 100.00% (13/13) |
|---------------------------------------------------+-----------------+-----------------+----------------+-----------------|
| src/lib/CriteriaResolution.sol                    | 100.00% (5/5)   | 100.00% (7/7)   | 100.00% (2/2)  | 100.00% (2/2)   |
|---------------------------------------------------+-----------------+-----------------+----------------+-----------------|
| test/foundry/utils/AgreementFrameworkTestBase.sol | 0.00% (0/40)    | 0.00% (0/49)    | 100.00% (0/0)  | 0.00% (0/9)     |
|---------------------------------------------------+-----------------+-----------------+----------------+-----------------|
| Total                                             | 61.21% (71/116) | 62.42% (93/149) | 52.50% (21/40) | 60.00% (15/25)  |
+---------------------------------------------------+-----------------+-----------------+----------------+-----------------+
```

```
forge coverage --report lcov
brew install lcov
genhtml lcov.info -o coverage
```
```
Generating output.
Processing file src/agreements/CollateralAgreement.sol
Processing file src/lib/CriteriaResolution.sol
Processing file test/foundry/utils/AgreementFrameworkTestBase.sol
Writing directory view page.
Overall coverage rate:
  lines......: 64.0% (71 of 111 lines)
  functions..: 62.5% (15 of 24 functions)
```
Screen Shot 2022-08-15 at 8 27 47 PM

### CODE REVIEW

**Review by johnmark13 — APPROVED** [review]

**Comment by aahna-ashina** [comment]
# `¯\_(ツ)_/¯`

Not sure why it says 46.85% at https://app.codecov.io/gh/nation3/court/pulls when `forge coverage` returns 64%.

Screen Shot 2022-08-15 at 9 12 22 PM

**Comment by johnmark13** [comment]
> # `¯\_(ツ)_/¯`
> Not sure why it says 46.85% at https://app.codecov.io/gh/nation3/court/pulls when `forge coverage` returns 63.96%.
> 
> > Screen Shot 2022-08-15 at 9 12 22 PM

Is that because codecov has branch coverage where it doesn't look like forge does??

**Review by gregfromstl — APPROVED** [review]
Looks good to me

**Comment by gregfromstl** [comment]
> > # `¯\_(ツ)_/¯`
> > Not sure why it says 46.85% at https://app.codecov.io/gh/nation3/court/pulls when `forge coverage` returns 63.96%.
> > > Screen Shot 2022-08-15 at 9 12 22 PM
> 
> Is that because codecov has branch coverage where it doesn't look like forge does??

I think they're both doing line coverage, but based on [this](https://docs.codecov.com/docs/about-code-coverage#:~:text=Unlike%20open%20source%20and%20paid,and%20bug%20fixes%20commonly%20occur.), codecov is considering partials as misses. It looks like forge is just counting partials as hits.
Screen Shot 2022-08-16 at 6 09 41 PM

**Comment by gregfromstl** [comment]
We should also add the `broadcast` directory to `.gitignore`

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,9 @@
 node_modules/
 cache/
 out/
-broadcast/
```
Comment (gregfromstl) [inline .gitignore:4]: Why remove this?

Comment (aahna-ashina) [reply .gitignore:4]: @gregfromstl I did not remove it:  https://github.com/nation3/court/pull/31/commits/46d49c060e3684355899bd335cb71438af8fc5bd#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947

I think that deletion happened when you merged the main branch.  Was there a merge conflict when you did the update?  https://github.com/nation3/court/commit/a738b1bea74b4de993bd10c3f83c496f52de9f84

**Review by xPi2 — CHANGES_REQUESTED** [review]
Remove test directory from the coverage please

**Comment by xPi2** [comment]
Feels great to have a high coverage of the contracts

**Comment by aahna-ashina** [comment]
> Remove test directory from the coverage please

@xPi2 I looked into it, and did some experimentation, but couldn't figure out how to exclude certain `.sol` files when running `forge coverage`.  So I asked for help here:  https://github.com/foundry-rs/foundry/issues/1961#issuecomment-1218870865

Will get back to you when he replies.

**Review by xPi2 — APPROVED** [review]

**Comment by aahna-ashina** [comment]
> Feels great to have a high coverage of the contracts, thanks @aahna-ashina for the initiative

Sure thing, @xPi2 😄

Also note that you can visit this page to get a quick overview of which lines have and haven't been covered:  https://app.codecov.io/gh/nation3/court/blobs/main/src/agreements/CollateralAgreement.sol

Screen Shot 2022-08-19 at 11 02 17 AM

**Pull request merged**